docusaurus/docs/dev-docs/providers.md

```diff
@@ -339,3 +368,58 @@ If you want to create your own provider without publishing it on npm you can fol
 
 4. Update your `./config/plugins.js` file to [configure the provider](#configuring-providers).
 5. Finally, run `yarn install` or `npm install` to install your new custom provider.
+
+## Private Providers
+
+You can configure a provider to be private, which means that asset URLs will be signed for secure access. To enable private providers, you need to implement the `isPrivate()` method and return `true`.
+
+### How It Works
+
+When the provider is set to private, every asset URL displayed in the Content Manager will be signed for secure access.
+
+In the backend, Strapi generates a signed URL for each asset using the `getSignedUrl(file)` method implemented in the provider. The signed URL includes an encrypted signature that allows the user to access the asset (but normally only for a limited time and with specific restrictions, depending on the provider).
+
+Note that for security reasons, the content API will not provide any signed URLs. Instead, developers using the API should sign the urls themselves.
+
+### Example
```

Comment:
```suggestion
**Example**
```

We tend to avoid generic titles like "How it works," "Understanding …" and "Example" and prefer them more 'meaningful' titles or avoid them. In this case, I understand you'd like to visually separate the example from the rest. Still, since we don't use any "Example" heading anywhere else in the doc, I'd recommend simply using bold formatting to make it stand out.

FYI if we had several examples, I'd recommend either using `Tabs` or `details` components for a cleaner view :)